filter.lua

```diff
@@ -123,20 +122,22 @@ local function filter(outbox)
 	end
 end
 
-local function sender(action, queue)
+local function sender(action, queue, event)
 	return function (...)
 		local outbox = mailbox.outbox(queue)
 		local ok, err = pcall(outbox.send, outbox, ...)
 		if not ok then
 			log("failed to %s: %s", action, err)
+		else
+			event:complete()
```

Comment:
sweet =)

Reply:
the `sender` design payed off quickly :>